creusot/src/analysis/init_locals.rs

```diff
@@ -93,6 +93,9 @@ where
                 | MutatingUseContext::Yield,
             ) => {}
 
+            // Ignore drops
+            PlaceContext::MutatingUse(MutatingUseContext::Drop) => {}
```

Comment:
this should be a `kill`. the local is not just ignored it is uninitialized after the drop.

Reply:
I checked what the original rustc pass did, https://github.com/rust-lang/rust/blob/80a96467ec5675e9f69683b5c075a8b15950c341/compiler/rustc_mir_dataflow/src/impls/init_locals.rs and it did seem like `Drop` was not a `kill`. I suppose this is a fine over approximation of unitialization, but normally when you combine this with the `maybeuninit` pass that should correctly rule out the cases that were causing problems above.

Reply:
if you're going to adjust `maybeinit` then you should also adjust `maybeuninit` in the same manner

Reply:
Also I do believe that it should be a `kill` here.